airflow/ui/src/components/Table.tsx

```diff
@@ -0,0 +1,182 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Custom wrapper of react-table using Chakra UI components
+*/
+
+import React, { useEffect } from 'react';
+import {
+  Flex,
+  Table as ChakraTable,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  chakra,
+  IconButton,
+  Text,
+  useColorModeValue,
+} from '@chakra-ui/react';
+import {
+  useTable, useSortBy, Column, usePagination, SortingRule,
+} from 'react-table';
+import {
+  MdArrowDropDown, MdArrowDropUp, MdKeyboardArrowLeft, MdKeyboardArrowRight,
+} from 'react-icons/md';
+
+interface Props {
+  data: any[];
+  columns: Column<any>[];
+  /*
+   * manualPagination is when you need to do server-side pagination.
+   * Leave blank for client-side only
+  */
+  manualPagination?: {
+    offset: number;
+    setOffset: (off: number) => void;
+    totalEntries: number;
+  };
+  /*
+   * setSortBy is for custom sorting such as server-side sorting
+  */
+  setSortBy?: (sortBy: SortingRule<object>[]) => void;
+  pageSize?: number;
+}
+
+const Table: React.FC<Props> = ({
+  data, columns, manualPagination, pageSize = 25, setSortBy,
+}) => {
+  const { totalEntries, offset, setOffset } = manualPagination || {};
+  const oddColor = useColorModeValue('gray.50', 'gray.900');
+  const hoverColor = useColorModeValue('gray.100', 'gray.700');
+
+  const pageCount = totalEntries ? (Math.ceil(totalEntries / pageSize) || 1) : data.length;
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    allColumns,
+    prepareRow,
+    page,
+    canPreviousPage,
+    canNextPage,
+    nextPage,
+    previousPage,
+    state: { pageIndex, sortBy },
+  } = useTable(
+    {
+      columns,
+      data,
+      pageCount,
+      manualPagination: !!manualPagination,
+      manualSortBy: !!setSortBy,
+      initialState: {
+        pageIndex: offset ? offset / pageSize : 0,
+        pageSize,
+      },
+    },
+    useSortBy,
+    usePagination,
+  );
+
+  const handleNext = () => {
+    nextPage();
+    if (setOffset) setOffset((pageIndex + 1) * pageSize);
+  };
+
+  const handlePrevious = () => {
+    previousPage();
+    if (setOffset) setOffset((pageIndex - 1 || 0) * pageSize);
+  };
+
+  useEffect(() => {
+    if (setSortBy) setSortBy(sortBy);
+  }, [sortBy, setSortBy]);
+
+  return (
+    <>
+      <ChakraTable {...getTableProps()}>
+        <Thead>
+          <Tr>
+            {allColumns.map((column) => (
+              <Th
+                {...column.getHeaderProps(column.getSortByToggleProps())}
+              >
+                {column.render('Header')}
+                <chakra.span pl="2">
```

Comment:
Can probably drop this span/padding—it looks a bit too distant from the heading text with it.